core/domain/topic_domain.py

```diff
@@ -1203,6 +1203,14 @@ def validate(self, strict: bool = False) -> None:
                 ' are not associated with the topic.' %
                 skill_ids_for_diagnostic_that_are_not_in_topic)
 
+        if (
+            strict and
+            len(self.skill_ids_for_diagnostic_test) == 0 and
+            len(all_skill_ids) > 0
```

Comment:
Is it actually possible for this (all_skill_ids) to have length 0 in strict mode?

It seems reasonable to me that published topics have at least one skill, but not sure if this is enforced. /cc @lkbhitesh07 @EricZLou in case they have any comments.

Reply:
Yes! You are correct. In the strict mode, there should be at least one subtopic attach to the Topic and also one skill attach to the subtopic, so I think it is not possible to have `all_skill_ids` length as 0.
Thanks

Reply:
Thank you, I have removed the 3rd check.